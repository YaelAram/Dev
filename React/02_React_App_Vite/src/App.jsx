import React from 'react';
import CounterApp from './components/Counter';
import Welcome from './components/Welcome';

/**
 * Everything that has no dependencies or hook from the component should be out of the function,
 * that improves the performance when React have to refresh the component
 */

function App(){
    return (
        <React.Fragment>
            <Welcome title="Hello World" subtitle="Goodbye World" message={ 123 }/>
        </React.Fragment>
    );
}

export default App;
