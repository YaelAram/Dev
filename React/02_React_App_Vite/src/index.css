body, input[type="button"]{
    background-color: #21232A;
    color: white;
}

body{
    padding: 2rem 1.5rem;
    letter-spacing: 1px;
    word-spacing: 1px;
    line-height: 1.2;
    font-family: Helvetica, Arial, sans-serif;
    text-align: center;
}

input[type="button"]{
    font-weight: bolder;
    font-size: large;
    border: 0.125rem solid white;
    border-radius: 0.5rem;
    padding: 1rem 1.5rem;
    margin: 0 0.5rem;
}

input[type="button"]:hover{
    background-color: white;
    color: #21232A;
}
