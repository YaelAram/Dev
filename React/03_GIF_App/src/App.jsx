import React, { Fragment, useState } from 'react';

import { SearchBar } from './components/SearchBar';
import { TopicSection } from './components/TopicSection';

export function App() {
    const [ topics, setTopics ] = useState( [] );

    const handleAddTopic = ( newTopic ) => {
        setTopics( ( prevTopics ) => {
            const index = prevTopics.findIndex( ( topic ) => topic === newTopic );
            if( index === -1 ) return [ newTopic, ...prevTopics ];
            else if( index === 0 ) return [ ...prevTopics ];
            else{
                const [ repeatedTopic ] = prevTopics.splice( index, 1 );
                return [ repeatedTopic, ...prevTopics ];
            }
        } );
    };

    const handleRemoveTopic = ( topicDelete ) => {
        const updatedTopics = topics.filter( ( topic ) => topicDelete !== topic );
        setTopics( updatedTopics );
    };

    return (
        <Fragment>
            <h1>React GIF App</h1>
            <SearchBar onAddTopic={ handleAddTopic } />
            { topics.map( ( topic ) => <TopicSection key={ topic } topic={ topic } onDeleteTopic={ handleRemoveTopic } /> ) }
        </Fragment>
    );
};
