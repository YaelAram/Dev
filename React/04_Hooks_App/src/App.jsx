import React from 'react';
import { MainApp } from './useContext/MainApp';

export function App() {
    return (
        <React.Fragment>
            <MainApp />
        </React.Fragment>
    );
};
