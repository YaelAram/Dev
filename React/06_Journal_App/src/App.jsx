import React from 'react';
import { AppRouter } from './router/AppRouter';

import { AppTheme } from './theme';

export function App() {
    return (
        <AppTheme>
            <AppRouter />
        </AppTheme>
    );
};
