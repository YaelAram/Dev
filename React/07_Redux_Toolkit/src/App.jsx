import React, { useEffect } from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { increment, decrement, incrementByAmount } from './store/slices/counter/counter';
import { getPokemon } from './store/slices/pokemon/thunks';

export function App() {
    const { value } = useSelector( state => state.counter );
    const { page, pokemon, isLoading } = useSelector( state => state.pokemon );
    const dispatch = useDispatch();

    useEffect( () => {
        dispatch( getPokemon( value ) );
    }, [ value ] );

    return (
        <React.Fragment>
            <h2>Counter: { value }</h2>
            <input type="button" value="Increment" onClick={ () => dispatch( increment() ) } />
            <input type="button" value="Decrement" onClick={ () => dispatch( decrement() ) } />
            <input type="button" value="Increment 2" onClick={ () => dispatch( incrementByAmount( 2 ) ) } />
            <hr />
            <h2>Pokemon</h2>
            <p>Loading Page { page }: { isLoading ? 'True' : 'False' }</p>
            <ul>
                {
                    pokemon.map( ( item ) => <li key={ item }>{ item }</li> )
                }
            </ul>
            <input type="button" value="Previous Page" disabled={ isLoading } onClick={ () => dispatch( decrement() ) } />
            <input type="button" value="Next Page" disabled={ isLoading } onClick={ () => dispatch( increment() ) } />
        </React.Fragment>
    );
};
