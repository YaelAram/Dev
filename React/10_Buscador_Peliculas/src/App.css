.flex-center {
  display: flex;
  justify-content: center;
}

.page {
  flex-direction: column;
  align-items: center;
  width: 100%;
}

form {
  align-items: center;
}

main {
  margin-top: 2rem;
  width: 100%;
}

.movies {
  list-style: none;
  margin: 0;
  padding: 0;

  display: grid;
  width: 100%;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  gap: 20px;
}

.movie {
  text-align: center;
}

.movie img {
  border-radius: 8px;
}

.custom-loader {
  width: 50px;
  height: 50px;
  --c: radial-gradient(farthest-side, #ffffff 92%, #0000);
  background: var(--c) 50% 0, var(--c) 50% 100%, var(--c) 100% 50%,
    var(--c) 0 50%;
  background-size: 10px 10px;
  background-repeat: no-repeat;
  animation: s8 1s infinite;
  position: relative;
}
.custom-loader::before {
  content: "";
  position: absolute;
  inset: 0;
  margin: 3px;
  background: repeating-conic-gradient(#0000 0 35deg, #ffffff 0 90deg);
  -webkit-mask: radial-gradient(farthest-side, #0000 calc(100% - 3px), #000 0);
  mask: radial-gradient(farthest-side, #0000 calc(100% - 3px), #000 0);
  border-radius: 50%;
}

@keyframes s8 {
  100% {
    transform: rotate(0.5turn);
  }
}
