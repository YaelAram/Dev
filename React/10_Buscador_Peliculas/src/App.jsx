import { useMemo } from "react";
import debounce from "just-debounce-it";

import { Movies, Loader } from "./components";
import { useMovies, useQuery } from "./hooks";

import "./App.css";

function App() {
  const { query, error, sort, updateInput, handleSort } = useQuery();
  const { hasMovies, movies, isLoading, errorMsg, getMovies } = useMovies({
    query,
    sort,
  });

  const debouncedGetMovies = useMemo(
    () =>
      debounce((query) => {
        getMovies(query);
      }, 500),
    [getMovies]
  );

  const handleChange = (evt) => {
    const newQuery = evt.target.value;
    updateInput(newQuery);
    debouncedGetMovies(newQuery);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    if (query) getMovies(query);
  };

  return (
    <div className="flex-center page">
      <header>
        <h1>Buscador de Peliculas</h1>
        <form className="flex-center" onSubmit={handleSubmit}>
          <input
            type="text"
            name="movie"
            placeholder="Star Wars"
            value={query}
            onChange={handleChange}
          />
          <input type="checkbox" onChange={handleSort} checked={sort} />
          <button type="submit">Search</button>
        </form>
        {error && <p style={{ color: "red" }}>{error}</p>}
      </header>
      <main className="flex-center">
        {isLoading ? (
          <Loader />
        ) : (
          <Movies hasMovies={hasMovies} movieTitle={query} movies={movies} />
        )}
        {errorMsg && <p style={{ color: "red" }}>{errorMsg}</p>}
      </main>
    </div>
  );
}

export default App;
