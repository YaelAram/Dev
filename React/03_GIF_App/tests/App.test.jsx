import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { App } from '../src/App';

describe( 'App component', () => {
    test( 'Snapshot', () => {
        const { container } = render( <App /> );
        expect( container ).toMatchSnapshot();
    } );

    test( 'should show the new topic without images', () => {
        const inputValue = 'rammstein';
        render( <App /> );
        const input = screen.getByRole( 'textbox' );
        const form = screen.getByRole( 'form' );
        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toBe( inputValue.toUpperCase() );
    } );

    test( 'should show the new topic with images', async () => {
        const inputValue = 'rammstein';
        render( <App /> );
        const input = screen.getByRole( 'textbox' );
        const form = screen.getByRole( 'form' );
        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );
        await waitFor( () => expect( screen.getAllByRole( 'img' ).length ).toBeGreaterThan( 0 ) );
        expect( screen.getAllByRole( 'img' ).length ).toBe( 6 );
    } );

    test( 'should delete TopicSection', async () => {
        const inputValue = 'rammstein';
        render( <App /> );
        const input = screen.getByRole( 'textbox' );
        const form = screen.getByRole( 'form' );
        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );
        await waitFor( () => expect( screen.getAllByRole( 'img' ).length ).toBeGreaterThan( 0 ) );
        const button = screen.getByRole( 'button' );
        fireEvent.click( button );
        expect( screen.getAllByRole( 'heading' ).length ).toBe( 1 );
    } );
} );
