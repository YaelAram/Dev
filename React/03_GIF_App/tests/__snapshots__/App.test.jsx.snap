// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot 1`] = `
<div>
  <h1>
    React GIF App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="e.g. Dr. House"
      type="text"
    />
  </form>
</div>
`;
